Automattic/wp-calypso
# Duplicate views: Redirect from Calypso to WP Admin (Posts, Testimonials, Portfolio)

Part of https://github.com/Automattic/wp-calypso/issues/95621
Part of https://github.com/Automattic/wp-calypso/issues/96183

## Proposed Changes

Introduces a new util to users participating in the "Remove duplicate views" experiment are redirected from Calypso to WP Admin if they try to access directly a Calypso screen of a removed view.

For now, this new logic only works with Posts, Testimonials, and Portfolio Projects.

https://github.com/user-attachments/assets/01215fa7-934d-49e2-b53b-285e5d4ea995


## Why are these changes being made?

Because we want to remove the custom Calypso implementations in favor of the standard WordPress WP Admin experience.

## Testing Instructions

- Go to the placeholder experiment page: 22124-explat-experiment
- Assign yourself to the `control` variation using the bookmarklets
- Use the Calypso live link below
- Go to `/posts/:site` for a site with the default admin interface
- Make sure you remain in Calypso
- Go to `/types/jetpack-testimonial/:site` for a site with the default admin interface
  - This requires Testimonials to be enabled, you can do that in Settings > Writing.
- Make sure you remain in Calypso
- Go to `/types/jetpack-testimonial/:site` for a site with the default admin interface
  - This requires Portfolio to be enabled, head to Settings > Writing and toggle their options.
- Make sure you remain in Calypso
- Go to the placeholder experiment page: 22124-explat-experiment
- Assign yourself to the `treatment` variation now
- Go to `/posts/:site` for a site with the default admin interface
- Make sure you are redirected to WP Admin
- Go to `/types/jetpack-testimonial/:site` for a site with the default admin interface
  - This requires Testimonials to be enabled, you can do that in Settings > Writing.
- Make sure you are redirected to WP Admin
- Go to `/types/jetpack-testimonial/:site` for a site with the default admin interface
  - This requires Portfolio to be enabled, head to Settings > Writing and toggle their options.
- Make sure you are redirected to WP Admin

## Pre-merge Checklist
